### ISSUE TITLE
Need a way to edit/set/reduce image size

### ISSUE THREAD

**Issue by xumx** [cjwirth/RichEditorView#31]
photos from insertImage can be huge. 
Is there anyway to change/set the display size to 50% width etc.

Something like what [Summernote](https://github.com/summernote/summernote) is doing. 

`style="width: 50%"`

**Comment by cjwirth** [comment]
Yeah, I agree, something like this should be in there. I'll see what I can whip up.

**Comment by markchous** [comment]
Sorry that I am super late on this, but you can always inject a viewport inside the meta tag of the html before setting it on the editor. That's how I solved this exact problem without losing quality of the image.

**Comment by swapnalishinde** [comment]
Hello guys, can anyone please explain in detail?

**Comment by swapnalishinde** [comment]
Got the solution but we need to modify some of the files:
![simulator screen shot 16-nov-2017 6 06 15 pm](https://user-images.githubusercontent.com/5267845/32891606-385ee314-caf9-11e7-92a1-61816626e37e.png)

**Comment by tnylee** [comment]
Any update on this?

**Comment by tnylee** [comment]
@markchous Can you please explain more on your solution? Thanks!

**Comment by sultansultanm97** [comment]
can you please share modified code

**Comment by pandomic** [comment]
You can do something like:

```swift
func insertImage(editor: RichEditorView, src: String, width: Int? = nil, height: Int? = nil, alt: String = "") {
    let finalWidth = width == nil ? "auto" : "\(width!)px"
    let finalHeight = height == nil ? "auto" : "\(height!)px"
    editor.html += "<img src=\"\(src)\" alt=\"\(alt)\" style=\"width: \(finalWidth); height: \(finalHeight); max-width: 100%;\" />"
}
```

Note that I've used the `max-width` constraint to not to overflow the max width of a web view, you may add/remove it on your own (or add a max-height as well)
